src/app/components/Modals/ErrorModal.tsx

```diff
@@ -71,6 +73,8 @@ export const ErrorModal: React.FC<ErrorModalProps> = ({
         return `Failed to stake due to: ${errorMessage}`;
       case ErrorState.WALLET:
         return `Failed to switch network due to: ${errorMessage}`;
+      case ErrorState.TERMS:
+        return `Failed to accept terms due to: ${errorMessage}`;
```

Comment:
```suggestion
        return `Failed to log terms acceptance due to: ${errorMessage}`;
```

These methods are related to the logging of the acceptance of the terms, not to the action of accepting the terms. We should clearly differentiate between that to avoid confusion to readers of the code.